src/main/java/ca/team2706/frc/robot/commands/drivebase/DriverAssistVision.java

```diff
@@ -14,189 +15,335 @@
+        generateTrajectoryRequestStageComplete = false;
+        ringLightOnDelayTime = 0.0;
+        ringLightOnStageComplete = false;
     }
 
     /**
      * Creates the driver assist command
      *
-     * @param driverAssistCargoAndLoading: True if target is cargo ship or loading bay, false otherwise
-     * @param driverAssistRocket:          True if target is a rocket ship, false otherwise
+     * @param DriverAssistVisionTarget target: The type of the destination target (CARGO_AND_LOADING, ROCKET, BALL)
      */
-    public DriverAssistVision(boolean driverAssistCargoAndLoading, boolean driverAssistRocket) {
-        this.driverAssistCargoAndLoading = driverAssistCargoAndLoading;
-        this.driverAssistRocket = driverAssistRocket;
+    public DriverAssistVision(DriverAssistVisionTarget target) {
+        this.target = target;
 
         // Ensure that this command is the only one to run on the drive base
-        requires(DriveBase.getInstance());
-
-        setupNetworkTables();
+        requires(RingLight.getInstance());
 
         commandAborted = false;
-        trajGenerated = false;
-        visionOffline = false;
-        angYawTargetWrtCameraLOSCWposPrev = 0.0;
-        distanceCameraToTarget_CameraPrev = 0.0;
+        generateTrajectoryRequestStageComplete = false;
+        ringLightOnDelayTime = 0.0;
+        ringLightOnStageComplete = false;
+        
+        setupNetworkTables();
     }
 
     /**
      * Sets up network tables used to get data from vision system
      */
     private void setupNetworkTables() {
-        // Set up network table
         inst = NetworkTableInstance.getDefault();
-        table = inst.getTable("PathFinder");
+        pathfinderTable = inst.getTable("PathFinder");
+        chickenVisionTable = inst.getTable("ChickenVision");
+        if ((pathfinderTable == null) || (chickenVisionTable == null)) {
+            Log.d("DAV: Network tables not found, command aborted");
+            commandAborted = true;
+        }
     }
 
     /**
-     * initialize() method does all the work of the command up to and including generating the trajectory
+     * initialize() method turns on ring light and sets up network tables
      */
     @Override
     public void initialize() {
-        // See method generateTrajectoryRobotToTarget(...) for an explanation of variable names and coodinate
-        // frames
-
         Log.d("DAV: initialize() called");
 
+        // If command was already aborted (e.g. due to errors in setup of network tables), 
+        // make sure we don't go any further
+        if (commandAborted) {
+            Log.d("DAV: Command aborted, initialize() not run");
+            return;
+        }
+
+        // Check if vision has stopped for some unexpected reason
+        if (visionOffline()) {
+            Log.d("DAV: Vision offline, command aborted");
+            commandAborted = true;
+            return;
+        }
+
         DriveBase.getInstance().setBrakeMode(true);
```

Comment:
Since DriverAssistVision isn't actually driving the robot, these lines should be removed

Reply:
Sorry again, @eandr127, but I don't know how to interpret the code snippet in this comment. Do you mean get rid of

`DriveBase.getInstance().setBrakeMode(true);` ?

Reply:
(Accidentally marked as resolve instead of replying)

Yes, just get rid of the brake mode and set position gyro calls.

Reply:
Removed.